grpc/internal/resolver/dns/dns_resolver_test.go

```diff
@@ -0,0 +1,42 @@
+package dns
```

Comment:
The upstream has a fair bit of testing code https://github.com/grpc/grpc-go/blob/v1.49.0/internal/resolver/dns/dns_resolver_test.go. It seems likely we could adapt some of that, particularly the SRV-handling stuff. Also I noticed you removed some of the interfaces that were presumably used for test mocks. We might end up wanting to put those back in.